console/src/components/Error/ErrorBoundary.tsx

```diff
@@ -79,21 +83,25 @@ class ErrorBoundary extends React.Component<
   };
 
   render() {
-    const { metadata } = this.props;
-    const { hasError, type, error } = this.state;
+    const { metadata, errorValue, requestError, requestURL } = this.props;
+    const { hasReactError, type, error } = this.state;
 
-    if (hasError && metadata.ongoingRequest) {
+    if (hasReactError && metadata.ongoingRequest) {
       return <Spinner />;
     }
 
-    if (hasError) {
+    if (hasReactError) {
       return type === '404' ? (
         <PageNotFound resetCallback={this.resetState} />
       ) : (
         <RuntimeError resetCallback={this.resetState} error={error} />
       );
     }
 
+    if (requestError===null && errorValue) {
+        return <ApiError error={errorValue} requestURL={requestURL} />
+    }
+
```

Comment:
I think in most cases, it should work well. But yes, we should try out as many different scenarios and make sure all requests are being made using `requestAction` and these errors are caught appropriately.